commit 5cb515a2592fefd70958d8b07efbb816645b6404 (refs/changes/58/2558/1)
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2017-09-13T13:44:10-06:00 (2 years, 1 month ago)
    
    MORAY-433 Bunyan serializer for postgres connection pool logs empty object

diff --git a/lib/manatee.js b/lib/manatee.js
index a7263b3..e87b128 100644
--- a/lib/manatee.js
+++ b/lib/manatee.js
@@ -49,13 +49,15 @@ var SERIALIZERS = {
         return (c ? c._moray_id : undefined);
     },
     pool: function (pool) {
-        var obj = {};
-
-        Object.keys(pool).forEach(function (k) {
-            obj[k] = pool[k] ? pool[k].url : undefined;
+        return ({
+            available: pool.available.length,
+            max: pool.max,
+            size: pool.resources.length,
+            waiting: pool.queue.length,
+            pending: pool.pendingResources,
+            stopped: pool.stopped,
+            stopping: pool.stopping
         });
-
-        return (obj);
     },
     db: function (db) {
         try {
diff --git a/lib/standalone.js b/lib/standalone.js
index 0e9e080..2a4a776 100644
--- a/lib/standalone.js
+++ b/lib/standalone.js
@@ -30,13 +30,15 @@ var SERIALIZERS = {
         return (c ? c._moray_id : undefined);
     },
     pool: function (pool) {
-        var obj = {};
-
-        Object.keys(pool).forEach(function (k) {
-            obj[k] = pool[k] ? pool[k].url : undefined;
+        return ({
+            available: pool.available.length,
+            max: pool.max,
+            size: pool.resources.length,
+            waiting: pool.queue.length,
+            pending: pool.pendingResources,
+            stopped: pool.stopped,
+            stopping: pool.stopping
         });
-
-        return (obj);
     },
     db: function (db) {
         var p = db.pool;
